commit b2bcc7b299f37037b4a78dc1538e5d6508ae8110
Author: Alok Kataria <akataria@vmware.com>
Date:   Fri Oct 31 11:59:53 2008 -0700

    x86: add a synthetic TSC_RELIABLE feature bit
    
    Impact: None, bit reservation only
    
    Add a synthetic TSC_RELIABLE feature bit which will be used to mark
    TSC as reliable so that we could skip all the runtime checks for
    TSC stablity, which have false positives in virtual environment.
    
    Signed-off-by: Alok N Kataria <akataria@vmware.com>
    Signed-off-by: Dan Hecht <dhecht@vmware.com>
    Signed-off-by: H. Peter Anvin <hpa@zytor.com>

Backported to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN a/include/asm-x86/cpufeature.h b/include/asm-x86/cpufeature.h
--- a/include/asm-x86/cpufeature.h	2008-07-13 15:51:29.000000000 -0600
+++ b/include/asm-x86/cpufeature.h	2009-07-16 14:25:56.000000000 -0600
@@ -79,6 +79,7 @@
 #define X86_FEATURE_REP_GOOD	(3*32+16) /* rep microcode works well on this CPU */
 #define X86_FEATURE_MFENCE_RDTSC (3*32+17) /* Mfence synchronizes RDTSC */
 #define X86_FEATURE_LFENCE_RDTSC (3*32+18) /* Lfence synchronizes RDTSC */
+#define X86_FEATURE_TSC_RELIABLE (3*32+23) /* TSC is known to be reliable */
 
 /* Intel-defined CPU features, CPUID level 0x00000001 (ecx), word 4 */
 #define X86_FEATURE_XMM3	(4*32+ 0) /* Streaming SIMD Extensions-3 */
